src/parse-stream.js

```diff
@@ -412,4 +417,28 @@ export default class ParseStream extends Stream {
       data: line.slice(4)
     });
   }
+
+  /**
+   * Add a parser for custom headers
+   *
+   * @param {RegExp}   expresion  a regular expression to match the custom header
+   * @param {String}   type       the custom type to register to the output
+   * @param {Function} dataParser function to parse the line into an object
+   */
+  addParser(expression, customType, dataParser) {
+    if (typeof dataParser != 'function') {
+      dataParser = (line) => line;
+    }
+    this.customParsers.push(line => {
+      const match = expression.exec(line);
```

Comment:
Not sure if our linter is here, but it may complain about no space after variable declaration